apps/medon-fe/src/components/SelectDoctor/index.tsx

```diff
@@ -76,31 +110,31 @@ export default function SelectDoctor({
         <ColumnName>{t('appointment.columns.located')}</ColumnName>
       </TitleBox>
       <List>
-        {mockDoctors.map((doctor) => (
-          <ListItem key={doctor.id}>
-            <ItemWrap
-              onClick={() => selectDoctor(doctor.id)}
-              style={isActiveDoc === doctor.id ? SlotActive : {}}
-            >
-              <ColumnText>
-                <DoctorPic
-                  src={doctor.photo || doctorImagePlaceholder}
-                  alt={`${t('appointment.doctorPicAlt')}`}
-                />
-                {t('appointment.prefix-doctor')}
-                {doctor.firstName[0]}. {doctor.lastName}
-              </ColumnText>
-              <ColumnText>
-                {specialityOptions.length
-                  ? specialityOptions[doctor.specialityId].label
-                  : ''}
-              </ColumnText>
-              <ColumnText>
-                {doctor.country}, {doctor.city}
-              </ColumnText>
-            </ItemWrap>
-          </ListItem>
-        ))}
+        {filteredDoctors
+          .filter((doctor: Doctor) => selectedDoctorsById.includes(doctor.id))
```

Comment:
you make filter with each rerender its O(n), not bad, but still heavy. find other way